app/src/lib/git/branch.ts

```diff
@@ -39,7 +39,14 @@ export async function deleteBranch(repository: Repository, branch: Branch, accou
 
   // If the user is not authenticated, the push is going to fail
   // Let this propagate and leave it to the caller to handle
-  if (remote) {
+  if (!remote) {
+    return true
+  }
+
+  const isBranchExistsOnRemote = await checkIfBranchExistsOnRemote(repository, branch, account, remote)
```

Comment:
Yup, long names are fine 👍 The only catch is that `isBranchExistsOnRemote` isn't grammatically correct. Mind changing it to `doesBranchExistOnRemote` or `branchExistsOnRemote` or something like that?